packages/components/src/internal/components/entities/EntityParentTypeSelectors.spec.tsx

```diff
@@ -0,0 +1,236 @@
+        List.of(
+            EntityParentType.create({ schema: 'a', query: 'test1', index: 1 }),
+            EntityParentType.create({ schema: 'a', query: 'test2', index: 2 })
+        )
+    );
+    return map;
+}, Map<string, List<EntityParentType>>());
+const EMPTY_OPTIONS_MAP = DATA_TYPES.reduce((map, type) => {
+    map = map.set(type.typeListingSchemaQuery.queryName, List());
+    return map;
+}, Map<string, List<IParentOption>>());
+
+describe('EntityParentTypeSelectors', () => {
+    const DEFAULT_PROPS = {
+        parentDataTypes: List<EntityDataType>(),
+        parentOptionsMap: Map<string, List<IParentOption>>(),
+        entityParentsMap: Map<string, List<EntityParentType>>(),
+        combineParentTypes: false,
+        onAdd: jest.fn,
+        onChange: jest.fn,
+        onRemove: jest.fn,
+    };
+
+    function validate(wrapper: ReactWrapper, inputCount = 0, addBtnCount = 0): void {
+        expect(wrapper.find('SelectInput')).toHaveLength(inputCount);
+        expect(wrapper.find('RemoveEntityButton')).toHaveLength(inputCount);
+        expect(wrapper.find('EntityParentTypeAddEntityButton')).toHaveLength(addBtnCount);
+    }
+
+    test('default props', () => {
+        const wrapper = mount(<EntityParentTypeSelectors {...DEFAULT_PROPS} />);
+        validate(wrapper);
+        wrapper.unmount();
+    });
+
+    test('parentDataTypes but no existing parents', () => {
+        const wrapper = mount(
+            <EntityParentTypeSelectors
+                {...DEFAULT_PROPS}
+                parentDataTypes={DATA_TYPES}
+                parentOptionsMap={EMPTY_OPTIONS_MAP}
+                entityParentsMap={EMPTY_PARENTS_MAP}
+            />
+        );
+        validate(wrapper, 0, 2);
+        wrapper.unmount();
+    });
+
+    test('combineParentTypes but no existing parents', () => {
+        const wrapper = mount(
+            <EntityParentTypeSelectors
+                {...DEFAULT_PROPS}
+                parentDataTypes={DATA_TYPES}
+                parentOptionsMap={EMPTY_OPTIONS_MAP}
+                entityParentsMap={EMPTY_PARENTS_MAP}
+                combineParentTypes
+            />
+        );
+        validate(wrapper, 0, 1);
+        wrapper.unmount();
+    });
+
+    test('parentDataTypes with existing parents', () => {
+        const wrapper = mount(
+            <EntityParentTypeSelectors
+                {...DEFAULT_PROPS}
+                parentDataTypes={DATA_TYPES}
+                parentOptionsMap={EMPTY_OPTIONS_MAP}
+                entityParentsMap={WITH_PARENTS_MAP}
+            />
+        );
+        validate(wrapper, 4, 2);
+
+        expect(wrapper.find('SelectInput').first().prop('label')).toBe('Parent 1 Type');
+        expect(wrapper.find('SelectInput').last().prop('label')).toBe('Source 2 Type');
+        expect(wrapper.find('RemoveEntityButton').first().prop('entity')).toBe('Parent');
+        expect(wrapper.find('RemoveEntityButton').last().prop('entity')).toBe('Source');
```

Comment:
I generally tend to prefer finding using the class instead of the string name these days as it is more reliable, particularly in the face of memoized components, but this clearly works, so no need to change.

Reply:
Ah yes, that's right. I forgot about that preference. Done.